src/people.rs

```diff
@@ -230,13 +323,18 @@ impl ContextPeopleExt for Context {
         }
 
         // Initialize the property. This does not fire a change event
-        let initialized_value = T::initialize(self, person_id);
+        // ok to call unwrap here because the property that returns None from get_person_property_ref
+        // in this case (i.e., hasn't been initialized yet) has a user-specified default/initializer
+        // and will return a some variant accordingly
+        let initialized_value = T::initialize(self, person_id).expect("Property not initialized");
```

Comment:
So while I think overall I'm going to recommend a different approach here, I wanted to give you some feedback on what would be a better way (IMO) to handle the mutability issues you're running into here –

What I would do here is have `include_in_periodic_report` be wrapped in a `RefCell` so that you could borrow it mutably without needing a mutable reference to context. That way get_person_property could still take an immutable reference.